sonaranalyzer-dotnet/src/Tests/SonarAnalyzer.UnitTest/TestCases/DebugAssertHasNoSideEffects.cs

```diff
@@ -0,0 +1,39 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void Test()
+        {
+            Debug.Assert2(true);
```

Comment:
If I understand this test correctly, perhaps we should use list.Contains("a") instead of "true".